clean_cost_report_1.py

```diff
@@ -0,0 +1,167 @@
+#####define function for extracting a given list of desired variables from the raw dataset
+
+def cost_data_reduce(costdata, varlist):
+    costdata['varname'] = ''
+    line_select = pd.Series(np.zeros(len(costdata))).astype(int)
+    for i in range(len(varlist[0])):
+        new_line_select = ((costdata['wksht_cd']==varlist[0][i]) & (costdata['line_num']==varlist[1][i]) & \
+                           (costdata['clmn_num']==varlist[2][i]))
+        costdata.loc[new_line_select, 'varname'] = varlist[3][i]
+        line_select = line_select | new_line_select ###TODO: might be able to make faster here
+
+    costdatasubset = costdata[['rep_rcd_num', 'varname', 'itm_val_num']][line_select].copy()
+    costdatasubset = costdatasubset.pivot(index='rep_rcd_num', columns='varname', values='itm_val_num')
+    costdatasubset.columns = [x for x in costdatasubset.columns]
+    costdatasubset.reset_index(inplace=True)
+
+    return costdatasubset
+
+
+
+#####make the lists of worksheet, line, and column numbers and variable names, for the 1996 version
+
+covar_list96 = []
+covar_list96.append(['S200000', 100, 100, 'city'])
+covar_list96.append(['S200000', 100, 200, 'state'])
+covar_list96.append(['S200000', 101, 300, 'zip'])
+covar_list96.append(['S200000', 101, 400, 'county'])
+
+covar_list96.append(['S200000', 200, 100, 'hosp_name'])
+covar_list96.append(['S200000', 200, 300, 'date_certified'])
+covar_list96.append(['S200000', 1900, 100, 'hosp_provider_type'])
+covar_list96.append(['S200000', 2101, 100, 'disproportionate_share'])
+covar_list96.append(['S200000', 2104, 100, 'geo_type_start'])
+covar_list96.append(['S200000', 2105, 100, 'geo_type_end'])
+covar_list96.append(['S200000', 2108, 100, 'method_medicaid'])
+covar_list96.append(['S200000', 2108, 200, 'method_medicaid_change'])
+
+covar_list96.append(['S200000', 2200, 100, 'referralcenter'])
+covar_list96.append(['S200000', 2300, 100, 'transplant'])
+covar_list96.append(['S200000', 2500, 100, 'teaching'])
+covar_list96.append(['S200000', 3000, 100, 'critical_access'])
+covar_list96.append(['S200000', 4700, 100, 'exemptionA'])
+covar_list96.append(['S200000', 4700, 200, 'exemptionB'])
+covar_list96.append(['S200000', 5400, 100, 'malpractice_prem'])
+covar_list96.append(['S200000', 5400, 200, 'malpractice_loss'])
+covar_list96.append(['S200000', 5800, 100, 'inpatient_rehab'])
+covar_list96.append(['S200000', 5900, 100, 'longterm_care_hosp'])
+covar_list96.append(['S200000', 6000, 100, 'inpatient_psych'])
+covar_list96.append(['S200000', 6100, 100, 'multicampus'])
+
+#convenient way to add variables with same root in name
+templines = [100, 500, 1200, 1400, 1700, 2500]
+tempnames = ['bed_'+x for x in ['hosp_adultped', 'total_adultped', 'total', 'subprovider', 'longterm', 'total_plus']]
+for i in range(len(templines)):
```

Comment:
This might be more readable as:
```
name_by_line = {
    100: "hosp_adultped",
    500: "total_adultped",
    ...
}
for line, name in name_by_line.items():
    covar_list96.append(['S300001', line, "bed_" + name])
```
Ditto for all the templines and tempnames below.